src/propulsion/abstract_motor_builder.hpp

```diff
@@ -0,0 +1,30 @@
+#ifndef SRC_PROPULSION_ABSTRACT_MOTOR_BUILDER_HPP_
+#define SRC_PROPULSION_ABSTRACT_MOTOR_BUILDER_HPP_
+
+#include <memory>
+#include "motor.hpp"
+#ifndef UNIT_TEST
+#include "propulsion_hardware_config.hpp"
+#else
+#include "propulsion_hardware_config_mock.hpp"
+#endif
+
+namespace propulsion
+
+{
+class AbstractMotorBuilder {
+ public:
+  AbstractMotorBuilder() = default;
+  virtual ~AbstractMotorBuilder() = default;
+  /**
+   * @brief Abstract method to construct a Motor object
+   * 
+   * @param motor_config The information by which it can be determined which Motor to build
+   * @return std::unique_ptr<Motor> A pointer to said Motor. Nullptr on error
+   */
+  virtual auto Create(propulsion::PropulsionHardwareConfig& motor_config) noexcept -> std::unique_ptr<Motor> = 0;
```

Comment:
Create method returns an unique pointer. This is not reflected in the UML model. Do we need that level of detail in the model or would you consider it irrelevant?

Reply:
I think it is fine that the model says, that it is a pointer. Do you see this differently?

Reply:
I agree! Ok, then we only have to set the correct type modifier in VP.

Reply:
@c0nwild Updated the diagram

Reply:
LGTM